api/src/db/DynamoBusinessesDataClient.ts

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { ExecuteStatementCommand, QueryCommand, QueryCommandInput } from "@aws-sdk/client-dynamodb";
+import { DeleteCommand, DynamoDBDocumentClient, GetCommand, PutCommand } from "@aws-sdk/lib-dynamodb";
+import { unmarshall } from "@aws-sdk/util-dynamodb";
+import { migrateData } from "@db/config/dynamoDbConfig";
+import { BusinessesDataClient } from "@domain/types";
+import { Business } from "@shared/userData";
+
+export const DynamoBusinessesDataClient = (
+  db: DynamoDBDocumentClient,
+  tableName: string
+): BusinessesDataClient => {
+  const doMigration = async (data: any): Promise<Business> => {
+    const migratedData = migrateData(data);
+    await put(migratedData);
+    return migratedData;
+  };
+
+  const findByIndex = (
+    indexName: string,
+    keyConditionExpression: string,
+    expressionAttributeValues: { [key: string]: any }
+  ): Promise<Business | undefined> => {
+    const params: QueryCommandInput = {
+      TableName: tableName,
+      IndexName: indexName,
+      KeyConditionExpression: keyConditionExpression,
+      ExpressionAttributeValues: expressionAttributeValues,
+    };
+
+    return db
+      .send(new QueryCommand(params))
+      .then((result) => {
+        if (!result.Items || result.Items.length !== 1) {
+          return;
+        }
+        const item = unmarshall(result.Items[0]);
+        return doMigration(item.data);
+      })
+      .catch((error) => {
+        console.log(error);
+        throw new Error("Not found");
+      });
+  };
+
+  const findByIndexArray = (
```

Comment:
[dust] `findAllByIndex`or `findBusinessByIndex` for line 21 and `findAllBusinessesByIndex` for the function above on line 48. `findByIndexArray` is just a tad confusing upon initial read, thoughts?